### DISCUSSION TITLE
VS Code debugging support for gdb on cortex-m7

### DISCUSSION THREAD

**Discussion by rsinventor** [japaric/discovery#420] [General]
I am going through section 5.3 in the Discovery book for the Microbit. I was wondering if there was any way to integrate gdb debugging into VS code for the Microbit. Any suggestions on how this could be done. What are the instructions to set up the debugger in VS Code? Please share.

**Comment by adamgreig** [comment]
Check out https://probe.rs/docs/tools/vscode/ for details of using probe-rs (which provides `cargo-embed` and GDB support) with VS code.

**Reply by rsinventor** [reply]
I tried to install the extension but I got this issue, [https://github.com/probe-rs/vscode/issues/24#issue-1087306584](https://github.com/probe-rs/vscode/issues/24#issue-1087306584).

**Reply by jemo07** [reply]
I followed this tutorial and it worked for me. 

https://www.justinmklam.com/posts/2017/10/vscode-debugger-setup/

Only gotcha is making sure the GDB you are running is compatible, I installed gdb-multiarch to ensure it supports ARM. 

Good luck!
